### PR TITLE
Fix JDK http client propagation of non-sampled traces

### PR DESCRIPTION
**Pull request by trask**
And adds missing test coverage to `HttpClientTest`

### CODE REVIEW

File: testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java [hunk 1]
```diff
@@ -195,6 +195,20 @@ void successfulRequestWithParent(String method) {
...
+    String method = "GET";
+    URI uri = resolveAddress("/success");
+    int responseCode = testing.runWithNonRecordingSpan(() -> doRequest(method, uri));
+
+    assertThat(responseCode).isEqualTo(200);
+
+    // sleep to ensure no spans are emitted
+    Thread.sleep(1000);
+
+    assertThat(testing.traces()).isEmpty();
```
Comment (tylerbenson) [inline testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java:209]: How does this verify that the proper headers are being sent?  Would you expect the server to generate a trace if it wasn't?

File: instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/HttpHeadersInstrumentation.java [hunk 2]
```diff
@@ -40,9 +39,7 @@ public void transform(TypeTransformer transformer) {
 
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void methodExit(@Advice.Return(readOnly = false) HttpHeaders headers) {
-      if (Java8BytecodeBridge.currentSpan().isRecording()) {
-        headers = tracer().inject(headers);
-      }
+      headers = tracer().inject(headers);
```
Comment (tylerbenson) [inline instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/HttpHeadersInstrumentation.java:42]: Looks like I just happened to pick an unlucky client. ;-)

**Review by tylerbenson — APPROVED** [review]

File: testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java (hunk 1 above)
Comment (trask) [reply testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java:209]: ya 👍

File: instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/HttpHeadersInstrumentation.java (hunk 2 above)
Comment (trask) [reply instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/HttpHeadersInstrumentation.java:42]: or a lucky one, to find this bug 💯

**Comment by trask** [comment]
closing and re-opening to re-trigger CLA check

File: testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java [hunk 3]
```diff
@@ -195,6 +195,20 @@ void successfulRequestWithParent(String method) {
...
+  @Test
+  void successfulRequestWithNotSampledParent() throws InterruptedException {
+    String method = "GET";
+    URI uri = resolveAddress("/success");
+    int responseCode = testing.runWithNonRecordingSpan(() -> doRequest(method, uri));
+
+    assertThat(responseCode).isEqualTo(200);
+
+    // sleep to ensure no spans are emitted
+    Thread.sleep(1000);
```
Comment (anuraaga) [inline testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java:207]: 1s seems too slow how about around 200? I think we get traces back in just a few ms most times.

**Review by anuraaga — APPROVED** [review]

**Review by iNikem — APPROVED** [review]

**Pull request merged**